### PR TITLE
fix VRFFluidTCtrl OU Fan Power not considering system cycling

### LINKED ISSUE #10390: VRF Cooling issue: Heat Pump outdoor unit fan power does not consider runtime fraction
Issue overview
--------------
When the heat pump runtime fraction is between 0 and 1, the outdoor unit fan power stays constant.

image

[eplusout_outdoor unit fan power not consider RTF.xlsx](https://github.com/NREL/EnergyPlus/files/14132684/eplusout_outdoor.unit.fan.power.not.consider.RTF.xlsx)

defect files

[US+SF+CZ4A+hp+crawlspace+IECC_2006_VRF.idf.txt](https://github.com/NREL/EnergyPlus/files/14132690/US%2BSF%2BCZ4A%2Bhp%2Bcrawlspace%2BIECC_2006_VRF.idf.txt)
[USA_NY_New.York-John.F.Kennedy.Intl.AP.744860_TMY3.epw.txt](https://github.com/NREL/EnergyPlus/files/14132699/USA_NY_New.York-John.F.Kennedy.Intl.AP.744860_TMY3.epw.txt)

### Details
Some additional details for this issue (if relevant):
 - Platform (Mac)
 - Version of EnergyPlus (23.2)
 - Unmethours link or helpdesk ticket number

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Defect file added (list location of defect file here)
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by yujiex**
Pull request overview
---------------------
 - Fixes #10390. 
 It was originally in PR #10341. Was separated out as its own separate PR.
 - currently the outdoor unit fan does not consider cycling. After the fix, the fan power is proportional to the system cycling ratio
image

### Regression diffs

Regression diffs occur in the following two files: 
- VariableRefrigerantFlow_FluidTCtrl_5Zone
- VariableRefrigerantFlow_FluidTCtrl_HR_5Zone

#### eplusout.csv diff
The diffs in eplusout.csv in the following variables.
- VRF HEAT PUMP:VRF Heat Pump Cooling Electricity Rate,
- VRF HEAT PUMP:VRF Heat Pump Cooling Electricity Energy,
- VRF HEAT PUMP:VRF Heat Pump Heating Electricity Rate,
- VRF HEAT PUMP:VRF Heat Pump Heating Electricity Energy,
- Cooling:Electricity,

VRF Heat Pump Cooling and Heating Electricity Rate and Energy differences are due to outdoor unit fan power changes in this feature (``this->ElecCoolingPower = state.dataHVACVarRefFlow->VRF(VRFCond).Ncomp + this->OUFanPower;`). 
Cooling:Electricity is affected by VRF Heat Pump Cooling/Heating Electricity, so the diffs are also expected.

#### meter diff
The following variables have meter diffs. These are expected as OU fan power is part of cooling, HVAC, and facility electricity. So these diffs are also expected.
- Cooling:Electricity
- Electricity:Facility 
- Electricity:HVAC

#### table diff

The following electricity terms have diffs. The Electricity:Facility, Electricity:HVAC, Heating:Electricity terms are due to reasons discussed above in meter and eplusout csv diffs. The ElectricityPurchased are also dependent on the electricity demand, which includes facility and HVAC electricity demand. These are affected by OU fan power as well. So diffs in *ElectricityPurchased is also expected.

- Electricity:Facility
- ElectricityPurchased:Facility
- ElectricityPurchased:Plant
- Cogeneration:ElectricityPurchased
- General:Cogeneration:ElectricityPurchased
- ElectricityNet:Facility
- ElectricityNet:Plant
- Cogeneration:ElectricityNet
- General:Cogeneration:ElectricityNet
- Electricity:HVAC
- Heating:Electricity
- General:Heating:Electricity
 
**NOTE: ENHANCEMENTS MUST FOLLOW A SUBMISSION PROCESS INCLUDING A FEATURE PROPOSAL AND DESIGN DOCUMENT PRIOR TO SUBMITTING CODE**

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: src/EnergyPlus/HVACVariableRefrigerantFlow.cc [hunk 1]
```diff
@@ -11736,7 +11736,7 @@ void VRFCondenserEquipment::CalcVRFCondenser_FluidTCtrl(EnergyPlusData &state)
         // Key outputs of this subroutine
         this->CompActSpeed = max(CompSpdActual, 0.0);
         this->Ncomp = max(Ncomp, 0.0) / this->EffCompInverter;
-        this->OUFanPower = this->RatedOUFanPower;
+        this->OUFanPower = this->RatedOUFanPower * CyclingRatio;
```
Comment (rraustad) [inline src/EnergyPlus/HVACVariableRefrigerantFlow.cc:11710]: This seems like a reasonable change IF the system uses cycling fan. For constant fan I don't think you want to do this.

Comment (rraustad) [reply src/EnergyPlus/HVACVariableRefrigerantFlow.cc:11710]: Oh, wait. this is the outdoor unit fan. Yes, I think this is reasonable that the entire system would cycle when below min part load ratio.

**Comment by dareumnam** [comment]
This is a side PR related to #10341, and it aligns with the changes we made in #10341. The test diffs are expected, and there are no conflicts with the develop branch. I believe this is good to go. @Myoldmopar

**Comment by Myoldmopar** [comment]
This is now causing a unit test failure in develop, at least locally.  I'm going to push the branch up and let CI confirm.  FWIW, I'm seeing the issue with the assertion that these two are equal-ish:

```
state->dataHVACVarRefFlow->VRF(1).OUFanPower
state->dataHVACVarRefFlow->VRF(1).RatedOUFanPower
```

**Comment by yujiex** [comment]
I will check the unit test

**Comment by Myoldmopar** [comment]
OK, CI confirmed my results, the unit test is failing here as seen here: https://github.com/NREL/EnergyPlus/actions/runs/10814479373/job/30001042106

**Comment by yujiex** [comment]
> OK, CI confirmed my results, the unit test is failing here as seen here: https://github.com/NREL/EnergyPlus/actions/runs/10814479373/job/30001042106

@Myoldmopar I just fixed it. Need to multiply by cycling ratio in unit test too.

**Comment by Myoldmopar** [comment]
Yep, that was it!  More diffs appearing, but obviously not because you changed one line in a unit test :)  It was behind develop and develop was moving way too fast.  Anyway, this is good to go, will merge it in later when CI calms down a bit.  Thanks!

**Comment by yujiex** [comment]
Thanks @Myoldmopar. Glad it resolved :)

**Comment by Myoldmopar** [comment]
OK, got the branch pulled and merged develop.  It's all happy now.  Thanks @yujiex !

**Pull request merged**